pages/en/contrib/style-guide-es6.md

```diff
@@ -0,0 +1,764 @@
+    // .
+    // .
+    // .
+    /** where are the tests, are we there yet? **/
+    // .
+    done();
+  });
+
+  beforeEach((done) => {
+    // reset
+  });
+
+  it('handles error like this');
+  it('handles error like that');
+ });
+```
+
+{% include note.html title="EXCEPTION:" content="Variables initialized by helpers and used from tests should be declared at the top of the `describe` block. Variables shared by multiple `describe`
+blocks may be placed in the outer scope.
+" %}
+
+For example:
+
+```js
+describe('my class', () => {
+  let app;
+  beforeEach(function setupApp);
+
+  it('does something');
+
+  function setupApp() {
+    app = loopback();
+    // ...
+  }
+});
+```
+### Callback function moved to the next line
+
+The following examples show the preferred style when callback needs to move to the next line due to line length exceeding max line length defined by eslint:
+
+**Good**:
+
+```js
+it('my long test description ...',
+function(done) {
+  ...
+});
+```
+
+**Bad**:
+
+```js
+it('my long test description ...',
+  function(done) {
+    …
+  });
+```
+
+**Bad**:
+
+```js
+it('my long test description ...',
+  function(done) {
+    …
+  }
+);
+```
+
+### Test naming
+
+Test names should describe the expected outcome under certain conditions. They should read like an English sentence, where `it` stands for the subject stated in `describe` argument.
+
+Use imperative mood, **do not** start test names with `should`.
+
+The test name should make clear:
+
+ - What is being tested, and the conditions specific to this test case.
+ - Expected outcome.
+
+Run `mocha -R spec` to review test names.
+
+**Good**:
+
+```js
+describe('strong-error-handler', () => {
+  it('returns status 500 by default');
+  // reads as: strong-error-handler returns status 500 by default
+ });
+
+describe('User', () => {
+  describe('login()', () => {
+    it('accepts valid credentials');
+    // reads as: User login() accepts valid credentials
+    it('creates an access token');
+    // reads as: User login() creates an access token
+  });
+);
+```
+
+**Bad**:
+
+```js
+describe('strong-error-handler', () => {
+  it('default status');
+  it('should return status 500 by default');
+});
+
+describe('User', () => {
+  describe('login()', () => {
+    it('works');
+    it('should create a token');
+  });
+});
+```
+
+### Test block naming
+
+Use `describe` for the top-level outer blocks and `context` for the inner
+blocks. `describe` should be used when we are describing the subject - what is
+being tested. In this particular example, both `Model` and `find()` should use
+`describe`. The goal is to create a human readable prefix to stand for `it` in the
+test cases.
+
+**Good**:
+
+```js
+describe('Model', () => {
+  describe('find()', () => {
+    context('with "include" filter', () => {
+      it('adds related models to the result', () => {
+        // Model find() returns filtered results
+      });
+    });
+  });
+
+  ...
+});
+```
+
+**Bad**:
```

Comment:
This `bad` code style snippet is the same as its `good counterpart above.

Reply:
Unless it's been changed, the differences in the codes are `describe` and `context`, same for the one down below

Reply:
I'll add a comment to both code examples.